### PR TITLE
prometheus: change threshold type value to float64

### LINKED ISSUE #2729: Prometheus Scaler threshold type should be float64 instead of int``
### Report

_No response_

### Expected Behavior

prometheus scaler should be able to accept decimal value, since prometheus metrics value is float64

### Actual Behavior

```
1.6469873430729506e+09	ERROR	controller.scaledobject	Failed to ensure HPA is correctly created for ScaledObject	{"reconciler group": "keda.sh", "reconciler kind": "ScaledObject", "name": "search-engine", "namespace": "default", "error": "error parsing prometheus metadata: error parsing threshold: strconv.Atoi: parsing \"0.5\": invalid syntax"}
```

### Steps to Reproduce the Problem

set a trigger with threshold in decimal value, for example `0.5`

### Logs from KEDA operator

```
example
```

### KEDA Version

2.6.1

### Kubernetes Version

1.21

### Platform

Amazon Web Services

### Scaler Details

Prometheus

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by alileza**
### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Fixes #2729

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -39,6 +39,7 @@
 - **Datadog Scaler:** Validate query to contain `{` to prevent panic on invalid query ([#2625](https://github.com/kedacore/keda/issues/2625))
 - **Kafka Scaler** Make "disable" a valid value for tls auth parameter ([#2608](https://github.com/kedacore/keda/issues/2608))
 - **RabbitMQ Scaler:** Include `vhost` for RabbitMQ when retrieving queue info with `useRegex` ([#2498](https://github.com/kedacore/keda/issues/2498))
+- **Prometheus Scaler:** Parse threshold as `float64` instead of `int` ([#2729](https://github.com/kedacore/keda/issues/2729))
```
Comment (JorTurFer) [inline CHANGELOG.md:42]: Could you sort this alphabetically please?

**Review by JorTurFer — CHANGES_REQUESTED** [review]
Thanks for your contribution! ❤️ 

I think that this change is not enough because the places where this value is used still cast it into int64. For supporting this feature, these lines have to be updated too:

- https://github.com/kedacore/keda/blob/main/pkg/scalers/prometheus_scaler.go#L147
- https://github.com/kedacore/keda/blob/main/pkg/scalers/prometheus_scaler.go#L244

The first line is when KEDA generates the HPA, that HPA should receive that threshold using mili.
The second line is when KEDA exposes the metric. If we round to int this value, doesn't make sense having float in threshold

**Comment by alileza** [comment]
@JorTurFer oh you're right, it wasn't covered by unit test, good catch I'll make the change!

**Comment by alileza** [comment]
this is interesting, https://github.com/kubernetes/apimachinery/blob/master/pkg/api/resource/quantity.go this dependency is not supporting float64

```
// Note that the quantity will NEVER be internally represented by a
// floating point number. That is the whole point of this exercise.
```

and if you see a few other scalers like [influxdb](https://github.com/kedacore/keda/blob/main/pkg/scalers/influxdb_scaler.go#L212), [huawei_cloudeye](https://github.com/kedacore/keda/blob/main/pkg/scalers/huawei_cloudeye_scaler.go#L237)

and a few others also facing the same issue. But casting float64 that would cause misbehaviour as well...

**Comment by JorTurFer** [comment]
Probably nobody has taken into account this...
Maybe I'm overthinking but without that change, I feel the support to float64 a bit useless because it's only internal. Nor HPA nor ScaledJobs will use the float for nothing

**Comment by zroubalik** [comment]
Yeah we should stick with the current implementation, it doesn't make sense to support float which will be immediately converted to integer.

Users can always multiply the prom query result and threshold with 100 to get higher precision.

**Pull request closed without merging**